endpoints/events/event.go

```diff
@@ -0,0 +1,363 @@
+
+	if err != nil {
+		w.WriteHeader(http.StatusUnauthorized)
+		w.Write([]byte(fmt.Sprintf("Account '%s' is required query parameter and can't be empty", AccountIdParameter)))
+		return
+	}
+	eventRequest.AccountID = accountId
+
+	if eventRequest.Analytics != Enabled {
+		return
+	}
+
+	ctx := context.Background()
+
+	// get account details
+	account, errs := GetAccount(ctx, e.Cfg, e.Accounts, eventRequest.AccountID)
+	if len(errs) > 0 {
+		w.WriteHeader(http.StatusInternalServerError)
+		for _, err := range errs {
+			w.Write([]byte(fmt.Sprintf("Internal Error: %s\n", err.Error())))
+		}
+
+		return
+	}
+
+	// account does not support events
+	if !account.EventsEnabled {
+		w.WriteHeader(http.StatusUnauthorized)
+		w.Write([]byte(fmt.Sprintf("Account '%s' doesn't support events", eventRequest.AccountID)))
+		return
+	}
+
+	// handle notification event
+	e.Analytics.LogNotificationEventObject(&analytics.NotificationEvent{
+		Type:      string(eventRequest.Type),
+		Timestamp: time.Unix(eventRequest.Timestamp, 0),
+		Bidid:     eventRequest.Bidid,
+		Bidder:    eventRequest.Bidder,
+		Account:   account,
+	})
+
+	// OK
+	w.WriteHeader(http.StatusOK)
+	// Add tracking pixel if format == image
+	if eventRequest.Format == Image {
+		w.Header().Add("Content-Type", e.TrackingPixel.ContentType)
+		w.Write(e.TrackingPixel.Content)
+	}
+}
+
+/**
+ * Converts an EventRequest to an URL
+ */
+func EventRequestToUrl(externalUrl string, request *EventRequest) string {
+	s := fmt.Sprintf(TemplateUrl, externalUrl, request.Type, request.Bidid, request.AccountID)
+
+	return s + optionalParameters(request)
+}
+
+/**
+ * Parses an EventRequest from an Http request
+ */
+func ParseEventRequest(r *http.Request) (*EventRequest, error) {
+	er := &EventRequest{}
+
+	// validate type
+	if err := validateType(er, r); err != nil {
+		return er, err
+	}
+
+	// validate bidid
+	if bidid, err := validateRequiredParameter(r, BidIdParameter); err != nil {
+		return er, err
+	} else {
+		er.Bidid = bidid
+	}
+
+	// validate timestamp (optional)
+	if err := validateTimestamp(er, r); err != nil {
+		return er, err
+	}
+
+	// validate format (optional)
+	if err := validateFormat(er, r); err != nil {
+		return er, err
+	}
+
+	// validate analytics (optional)
+	if err := validateAnalytics(er, r); err != nil {
+		return er, err
+	}
+
+	// Bidder
+	bidder := r.FormValue(BidderParameter)
+
+	if bidder != "" {
+		er.Bidder = bidder
+	}
+
+	return er, nil
+}
+
+func optionalParameters(request *EventRequest) string {
+	r := url.Values{}
+
+	// timestamp
+	if request.Timestamp > 0 {
+		r.Add(TimestampParameter, strconv.FormatInt(request.Timestamp, 10))
+	}
+
+	// bidder
+	if request.Bidder != "" {
+		r.Add(BidderParameter, request.Bidder)
+	}
+
+	// format
+	switch request.Format {
+	case Blank:
+		r.Add(FormatParameter, string(Blank))
+	case Image:
+		r.Add(FormatParameter, string(Image))
+	}
+
+	//analytics
+	switch request.Analytics {
+	case Enabled:
```

Comment:
Should this be the default ? My understand of this flag is that we can send x=0 to disable analytics, which seems to imply that a missing parameter means that it's enabled.